docs/rfc/rfc-6.adoc

```diff
@@ -0,0 +1,228 @@
+:toc: macro
+
+= RFC 6: Transaction incentives
+
+:icons: font
+:numbered:
+toc::[]
+
+== Goal
+The goal of this proposal is to reduce on-chain cost and complexity related to
+transaction submitter responsibilities, reduce the cost of operating a client,
+and provide tools to punish free-riders not operating their nodes properly.
+
+== Overview
+The scheme assumes that the majority of the group is honest and is interested in
+helping DAO to eliminate free-riders from the network while getting nothing in
+return. The group will vote to mark someone as ineligible for rewards if they
+consistently do not participate in off-chain heartbeats or if they skip their
+responsibilities for submitting on-chain transactions. 
+
+A real-world analogy would be team members discussing one of them is not
+contributing to the work at all so the entire team must do the work of that
+free-rider and have the team report this fact to the manager.
+
+All the responsibilities that do not have to be done by a group will be
+delegated to opt-in maintainers and third parties to reduce the cost of
+operating a node.
+
+== In depth
+
+=== Transaction types
+
+On-chain transactions can be divided into three types:
+
+- Transactions that can only be submitted by group members, given that only the
+  group members have the knowledge about input parameters. For example,
+  <<submit-dkg-result,submitting DKG result transaction>> for ECDSA wallet or
+  submitting relay entry transaction for the random beacon.
+  
+- Transactions that can be submitted by anyone, usually submitted often and
+  mostly belonging to a happy path. This type of transaction should never lead
+  to slashing. For example, <<maintainer-incentives,submit sweep proof or submit
+  redemption proof transaction>> for tBTC v2. 
+
+- Transactions that can be submitted by anyone, submitted rarely or never, and
+  usually leading to slashing. For example, <<challenge-dkg-result, challenge
+  DKG result>> for ECDSA wallet or <<report-fraud, report fraud>> transaction for
+  tBTC v2.
+
+For the first group of transactions, we will not enforce the order on-chain and
+we will expect off-chain clients to respect the submission order. If some client
+skipped their round, other clients can vote to mark the inactive client as
+ineligible for rewards. To disincentivize frontrunning, we will only reimburse
```

Comment:
I'm not 100% sure, but my understanding is that we need to implement and ETH pool described [here](https://github.com/keep-network/tbtc-v2/blob/98c2f6890d384e3f14833fb30635d843af73f666/docs/rfc/rfc-6.adoc#32-eth-pool) and reimbursement will happen upon calling a function on a contract that notifies about failed hearbeats. Something like [here](https://github.com/keep-network/keep-core/blob/main/solidity/random-beacon/contracts/RandomBeacon.sol#L944) but instead of transferring T tokens, we will need to reimburse ETH. Is it way off, Piotr?

Reply:
Implementation detail.